numpy/_core/src/umath/clip.cpp

```diff
@@ -196,6 +196,69 @@ _npy_clip(char **args, npy_intp const *dimensions, npy_intp const *steps)
     return _npy_clip_<Tag>((T **)args, dimensions, steps);
 }
 
+template <class Tag, class T = typename Tag::type>
+static void
+_npy_clip_floating_(T **args, npy_intp const *dimensions, npy_intp const *steps)
+{
+    npy_intp n = dimensions[0];
+    if (steps[1] == 0 && steps[2] == 0) {
+        /* min and max are constant throughout the loop, the most common case
+         */
+        T min_val = *args[1];
+        T max_val = *args[2];
+
+        T *ip1 = args[0], *op1 = args[3];
+        npy_intp is1 = steps[0] / sizeof(T), os1 = steps[3] / sizeof(T);
+
+        if (!npy_isnan(min_val) && !npy_isnan(max_val)) {
+            /* min_val and max_val are not nans */
+            /* nans in the input will be propagated naturally */
+
+            /* contiguous, branch to let the compiler optimize */
```

Comment:
Is this branch faster (and by how much)? Otherwise we could drop it to keep the code smaller

Reply:
I see now the other methods have the same branch. For consistency we can then keep it the same.